### PR TITLE
Fixed problems with a types and refactored the code of files sizes conversions

### PR DESCRIPTION
**Pull request by christianrss**
Fixed some problems like "A non-numeric value encountered" which occurs with a values without explicit type conversion. And refactored the code of file sizes conversions with a better approach. In this branch the field "Maximum size of uploads" presented problems with float numbers what has been fixed with the refactoring.

### CODE REVIEW

File: qa-include/qa-base.php [hunk 1]
```diff
@@ -1284,16 +1284,7 @@ function qa_post_limit_exceeded()
...
-		case 'g':
-			return $value * 1073741824;
-		case 'm':
-			return $value * 1048576;
-		case 'k':
-			return $value * 1024;
-		default:
-			return $value;
-	}
+	return intval($value) * 1024 ** (int) array_search(strtolower($unit), ['', 'k', 'm', 'g']);
```
Comment (svivian) [inline qa-include/qa-base.php]: This feels like it's trying to be too clever, and it makes the code very non-obvious. I know the current code uses somewhat "magic" numbers but it's simple and clear.

File: qa-plugin/wysiwyg-editor/qa-wysiwyg-editor.php [hunk 2]
```diff
@@ -113,9 +113,9 @@ public function admin_form(&$qa_content)
 				array(
 					'id' => 'wysiwyg_editor_upload_max_size_display',
 					'label' => 'Maximum size of uploads:',
-					'suffix' => 'MB (max '.qa_html(qa_format_number($this->bytes_to_mega(qa_get_max_upload_size()))).')',
+					'suffix' => 'MB (max '.qa_html(number_format($this->bytes_to_mega(qa_get_max_upload_size()), 1)).')',
 					'type' => 'number',
-					'value' => qa_html(number_format($this->bytes_to_mega(qa_opt('wysiwyg_editor_upload_max_size')))),
+					'value' => qa_html(number_format($this->bytes_to_mega(qa_opt('wysiwyg_editor_upload_max_size')), 1)),
```
Comment (svivian) [inline qa-plugin/wysiwyg-editor/qa-wysiwyg-editor.php]: Is there a reason to have decimals here?

**Comment by svivian** [comment]
This also doesn't appear to have the change from your previous PR.

File: qa-plugin/wysiwyg-editor/qa-wysiwyg-editor.php (hunk 2 above)
Comment (christianrss) [inline qa-plugin/wysiwyg-editor/qa-wysiwyg-editor.php]: Before this change it was not possible put float numbers in maximum file size, for example 1.5 MB.

File: qa-include/app/format.php [hunk 3]
```diff
@@ -1206,7 +1206,7 @@ function qa_html_page_links($request, $start, $pagesize, $count, $prevnext, $par
 	if (qa_to_override(__FUNCTION__)) { $args=func_get_args(); return qa_call_override(__FUNCTION__, $args); }
 
 	$thispage = 1 + floor($start / $pagesize);
-	$lastpage = ceil(min($count, 1 + QA_MAX_LIMIT_START) / $pagesize);
+	$lastpage = ceil(min((int)$count, 1 + QA_MAX_LIMIT_START) / $pagesize);
```
Comment (christianrss) [inline qa-include/app/format.php]: Here is the correction sent in the previous PR

**Comment by svivian** [comment]
Sorry for the long delay, finally got around to merging this (needed to fix some whitespace issues so I squashed to one commit). Thanks for the work!

**Pull request closed without merging**